flink-core/src/main/java/org/apache/flink/api/common/state/FoldingStateDescriptor.java

```diff
@@ -55,7 +55,7 @@
 	 * @param typeClass The type of the values in the state.
 	 */
 	public FoldingStateDescriptor(String name, ACC initialValue, FoldFunction<T, ACC> foldFunction, Class<ACC> typeClass) {
-		super(name, typeClass, initialValue);
+		super(name, typeClass, DefaultValueFactory.create(typeClass, requireNonNull(initialValue)));
 		this.foldFunction = requireNonNull(foldFunction);
```

Comment:
I think use `Preconditions. checkNotNull()` instead of `requireNonNull()` here is more consistence with the previous commit: `Make State Descriptors consistently use Preconditions…`.

Reply:
I had to remove the `checkNotNull()` all together, because `null` is actually a valid initial value.